org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/NumberTestCase.xtend

```diff
@@ -44,5 +44,177 @@ class NumberTestCase extends AbstractWollokInterpreterTestCase {
 		}
 		'''.interpretPropagatingErrors
 	}
+
+	@Test
+	def void integersBetwenTrue() {
+		'''program a {
+		
+			assert.that(3.between(1, 5))
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void integersBetweenFalse() {
+		'''program a {
+		
+			assert.notThat(3.between(5, 9))
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void absoluteValueOfAPositiveInteger() {
+		'''program a {
+		
+			assert.equals(3, 3.abs())
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void absoluteValueOfANegativeInteger() {
+		'''program a {
+		
+			assert.equals(3, (-3).abs())
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void squareRoot() {
+		'''program a {
+		
+			assert.equals(3, 9.squareRoot())
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void square() {
+		'''program a {
+		
+			assert.equals(9, 3.square())
+		
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void lessThan() {
+		'''program a {
+		
+			assert.that(3 < 7)
+			assert.notThat(13 < 7)
+		}
+		'''.interpretPropagatingErrors
+	}
 	
+	@Test
+	def void remainder() {
+		'''program a {
+			const remainder = 17 % 7
+			const remainder2 = 12 % 6
+			assert.equals(3, remainder)
+			assert.equals(0, remainder2)
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void even() {
+		'''program a {
+			assert.notThat(3.even())
+			assert.that(260.even())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void odd() {
+		'''program a {
+			assert.that(3.odd())
+			assert.notThat(260.odd())
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void gcd() {
+		'''program a {
+			const gcd1 = 5.gcd(25)
+			const gcd2 = 25.gcd(20)
+			const gcd3 = 2.gcd(3)
+			assert.equals(5, gcd1)
+			assert.equals(5, gcd2)
+			assert.equals(1, gcd3)
+		}
+		'''.interpretPropagatingErrors
+	}
+
+	@Test
+	def void gcdForDecimalsIsInvalid() {
+		try {
+		'''program a {
+			(5.5).gcd(12)
+		}
+		'''.interpretPropagatingErrors
+			fail("GCD is UNDEFINED for BigDecimals")
+		} catch (AssertionError e) {
+			assertTrue(e.message.startsWith("5.5 does not understand gcd(p0)"))
+		}
+	}
+
+	@Test
+	def void gcdForDecimalsIsInvalid2() {
+		try {
+		'''program a {
+			console.println((5).gcd(12.3))
+		}
+		'''.interpretPropagatingErrors
+			fail("You cannot pass a BigDecimal as first parameter in GCD")
+		} catch (AssertionError e) {
```

Comment:
Isn" t it a bit suprising to throw a sich  different error in this case?

It is talking about java and casting, instead of messages